source/constructs/api/data_source/service.py

```diff
@@ -2080,3 +2084,52 @@ def query_full_provider_resource_infos():
 
 def list_providers():
     return crud.query_provider_list()
+
+
+def get_schema_from_url(url):
+    res = ''
+    # jdbc:mysql://81.70.179.114:9000/sdps-glue
+    if url.startswith("jdbc:mysql://"):
+        res = url[url.rindex("/") + 1:]
+    return res
+
+
+def gen_jdbc_provider_str(provider: int):
+    if (provider == 2):
+        return DatabaseType.JDBC_TENCENT.value
+    elif (provider == 3):
+        return DatabaseType.JDBC_ALIYUN.value
+    else:
+        return DatabaseType.JDBC_AWS.value
+
+
+def grant_lake_formation_permission(credentials, crawler_role_arn, glue_database_name):
+    lakeformation = boto3.client('lakeformation',
+                                 aws_access_key_id=credentials['AccessKeyId'],
+                                 aws_secret_access_key=credentials['SecretAccessKey'],
+                                 aws_session_token=credentials['SessionToken'],
+                                 region_name=_admin_account_region)
+    """ :type : pyboto3.lakeformation """
+    # retry for grant permissions
+    num_retries = GRANT_PERMISSIONS_RETRIES
+    while num_retries > 0:
+        try:
+            response = lakeformation.grant_permissions(
+                Principal={
+                    'DataLakePrincipalIdentifier': f"{crawler_role_arn}"
+                },
+                Resource={
+                    'Database': {
+                        'Name': glue_database_name
+                    }
+                },
+                Permissions=['ALL'],
+                PermissionsWithGrantOption=['ALL']
+            )
+        except Exception as e:
+            sleep(SLEEP_MIN_TIME)
+            num_retries -= 1
+        else:
+            break
+    else:
+        raise Exception('UNCONNECTED')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.